### ISSUE TITLE
[FEATURE] Allow undefined environment values.

### ISSUE THREAD

**Issue by kareem-abdul** [vara855/yaml-env-defaults#74]
**Is your feature request related to a problem? Please describe.**
So I have a setup where parts of my config are optional. ie the env variables are not required.
For example, Consider the following yaml file.
```yaml
app:
   type: ${TYPE}
   feature1:
        someFieldForFeature1:  ${SOME_FIELD_FOR_FEATURE1}
   feature2:
         someFeildForFeature2: ${SOME_FIELD_FOR_FEATURE2} 
```
Here based on the value of `app.type` I'm validating either `app.feature1` or `app.feature2`  after the yaml was parsed.
So when my env look likes following
```env
TYPE=feature1
SOME_FIELD_FOR_FEATURE1=value
```
I'll get an error 
`Unknown environment variable referenced in config: SOME_FIELD_FOR_FEATURE2. Specify environment variable or use defaults syntax.`.

**Describe the solution you'd like**
Maybe add a config flag to allow this which would be by default false.
```ts
const yaml = readYamlEnvSync('config.yaml', process.env, { 
    allowUndefinedEnv: true,
});
```

**Describe alternatives you've considered**
I wrote the functions from scratch using `js-yaml` not really useful.

**Comment by kareem-abdul** [comment]
> Hi. @kareem-abdul thank you for the issue. Have you tried to provide a `getProperty` function? You can write your own logic for `undefined` case.

Hi @vara855 , I'm assuming you meant this,
```ts
const yaml = readYamlEnvSync('config.yaml', (key) => process.env[key]);
```
Unfortunately, this does not work either.

**Comment by vara855** [comment]
Hi, @kareem-abdul. I've made changes, you can check that [playground](https://stackblitz.com/edit/stackblitz-starters-ntdt2v?file=index.js) 

fixed in v2.0.5
